cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperatorTest.java

```diff
@@ -157,25 +762,23 @@ public void testNewRebalance(Vertx vertx, VertxTestContext context) throws IOExc
         when(mockRebalanceOps.updateStatusAsync(any(KafkaRebalance.class))).thenAnswer(invocation -> {
             try {
                 return Future.succeededFuture(Crds.kafkaRebalanceOperation(client)
-                        .inNamespace(CLUSTER_NAMESPACE)
-                        .withName(RESOURCE_NAME)
+                        .inNamespace(namespace)
+                        .withName(resource)
+                        .patch(invocation.getArgument(0)));
+            } catch (Exception e) {
+                return Future.failedFuture(e);
+            }
+        });
+        when(mockRebalanceOps.patchAsync(any(KafkaRebalance.class))).thenAnswer(invocation -> {
+            try {
+                return Future.succeededFuture(Crds.kafkaRebalanceOperation(client)
+                        .inNamespace(namespace)
+                        .withName(resource)
                         .patch(invocation.getArgument(0)));
             } catch (Exception e) {
                 return Future.failedFuture(e);
             }
         });
-
-        kcrao.reconcileRebalance(
-                new Reconciliation("test-trigger", KafkaRebalance.RESOURCE_KIND, CLUSTER_NAMESPACE, RESOURCE_NAME),
-                kcRebalance).setHandler(context.succeeding(v -> context.verify(() -> {
-                    KafkaRebalance kr = Crds.kafkaRebalanceOperation(client).inNamespace(CLUSTER_NAMESPACE).withName(RESOURCE_NAME).get();
-                    assertThat(kr, notNullValue());
-                    assertThat(kr.getStatus(), notNullValue());
-                    assertThat(kr.getStatus().getConditions(), notNullValue());
-                    assertThat(kr.getStatus().getConditions().get(0).getType(), is(KafkaRebalanceStatus.REBALANCE_STATUS_CONDITION_TYPE));
-                    assertThat(kr.getStatus().getConditions().get(0).getStatus(), is(KafkaRebalanceAssemblyOperator.State.ProposalReady.toString()));
-                    context.completeNow();
-                })));
     }
 
 }
```

Comment:
Overall it feels like you have good coverage of various paths through the states, which is good. You don't have any coverage for error cases (like what happens if CC returns an error). Could you add some tests for that?

How does the CC client cope with CC not responding, does it timeout the returned `Future`s (in which case that would be covered by the extra testing I just described)? If not we should probably plan to add that, as I think the operator is complex enough, and it's easier to add that to the client.

Reply:
@ppatierno can you comment on this one?

Reply:
So yes, currently the CC client uses the Vertx HTTP client but leaving the timeout to the default (which should be 60 secs). For testing, we should add a way to set a lower timeout on the CC client API.